### PR TITLE
Mixpanel destination plugin

### PR DESCRIPTION
**Pull request by pauloouriques**

### CODE REVIEW

File: plugins/@grouparoo/mixpanel/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+Learn more about Grouparoo Plugins at [www.grouparoo.com/docs/plugins](https://www.grouparoo.com/docs/plugins).
+
+## Installation
+
+In your Grouparoo project, run `grouparoo install @grouparoo/mixpanel`.
+
+## Attribution
+
+The Mixpanel icon used in this plugin is from https://livesession.io/integrations/mixpanel/
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/README.md]: Probably best to get the logo from somewhere else

File: plugins/@grouparoo/mixpanel/__tests__/export/export-profile.ts [hunk 2]
```diff
@@ -0,0 +1,879 @@
...
+const otherName = "Lucas";
+const newEmail = "carlos.solimoes@mailinator.com";
+const newName = "Carlos";
+const listOne = "List One";
+const listTwo = "List Two";
+const listThree = "List Three";
+const listFour = "List Four";
+
+const ltv = 3039;
+const hugeTime = helper.longTime * 6;
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/__tests__/export/export-profile.ts:35]: Ooof.

File: plugins/@grouparoo/mixpanel/__tests__/utils/nockHelper.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    nockCall = nockCall.replace(
+      new RegExp(realEnv.MIXPANEL_SECRET, "gi"),
+      nockEnv.MIXPANEL_SECRET
+    );
+    nockCall = nockCall.replace(
+      new RegExp(realEnv.MIXPANEL_PROJECT_ID, "gi"),
+      nockEnv.MIXPANEL_PROJECT_ID
+    );
+    return nockCall;
+  },
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/__tests__/utils/nockHelper.ts:51]: 👍

File: plugins/@grouparoo/mixpanel/package.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
+{
+  "author": "Grouparoo Inc <hello@grouparoo.com>",
+  "name": "@grouparoo/mixpanel",
+  "description": "The Grouparoo Mixpanel Plugin",
+  "version": "0.5.2-alpha.1",
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/package.json]: 👍

File: plugins/@grouparoo/mixpanel/package.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  "dependencies": {
+    "@grouparoo/app-templates": "0.5.2-alpha.1",
+    "axios": "0.21.1"
+  },
+  "devDependencies": {
+    "@grouparoo/core": "0.5.2-alpha.1",
+    "@grouparoo/spec-helper": "0.5.2-alpha.1",
+    "@types/jest": "*",
+    "@types/node": "*",
+    "actionhero": "26.1.2",
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/package.json:39]: Bumped to 26.1.2 now

File: plugins/@grouparoo/mixpanel/public/templates/destination/{{{id}}}.js.template [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+      options: {},
+
+      // Mappings are how you choose which properties to export to this destination.
+      // Keys are the name to display in the destination, values are the IDs of the Properties in Grouparoo.
+      mapping: {
+        $email: "email",
+        $first_name: "firstName",
+        $last_name: "lastName",
+      },
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/public/templates/destination/{{{id}}}.js.template]: It looks like `$distinctId` is required.  Can that be in the template please?

File: plugins/@grouparoo/mixpanel/src/lib/client/profile.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      url: `/api/2.0/engage`,
+      params: { project_id: this.client.projectId },
+      data: { distinct_id: distinctId },
+      baseURL: this.baseUrl,
+    });
+    if (response?.results?.length > 0) {
+      return response.results[0];
+    }
+    return null;
+  }
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/src/lib/client/profile.ts:24]: It's fascinating they don't have this method in their "public" API client - I guess they assumed all user management would be though their site...

File: plugins/@grouparoo/mixpanel/src/lib/export/destinationMappingOptions.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    return {
+      labels: {
+        property: {
+          singular: "Mixpanel Property",
+          plural: "Mixpanel Properties",
+        },
+        group: {
+          singular: "Mixpanel Groups",
+          plural: "Mixpanel Groups",
+        },
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/src/lib/export/destinationMappingOptions.ts]: Our terms match!

File: plugins/@grouparoo/mixpanel/src/lib/test.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { TestPluginMethod } from "@grouparoo/core";
+import { connect } from "./connect";
+
+export const test: TestPluginMethod = async ({ appOptions }) => {
+  const client = await connect(appOptions);
+  const result = await client.query.profile.getSavedCohorts();
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/src/lib/test.ts:6]: For a test method, will this always work... even if you have no saved cohorts?

**Review by evantahler — CHANGES_REQUESTED** [review]
This looks really good! Well done!  
I'm sorry that reading from the API was so annoying.

There are some minor things to change up listed below, but overall, this is great!

File: plugins/@grouparoo/mixpanel/src/lib/export/exportProfile.ts [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    if (syncOperations.update) {
+      // clear groups
+      await client.ingestion.profile.update(profile["$distinct_id"], {
+        groups: [],
+      });
+    }
+  }
+};
+
+export const exportProfile: ExportProfilePluginMethod = async ({
```
Comment (evantahler) [inline plugins/@grouparoo/mixpanel/src/lib/export/exportProfile.ts:26]: It's nice to not have an API rate limit, and that we can do the "simpler" thing of just worrying about 1 profile at a time.

File: plugins/@grouparoo/mixpanel/src/lib/client/profile.ts (hunk 7 above)
Comment (pauloouriques) [reply plugins/@grouparoo/mixpanel/src/lib/client/profile.ts:24]: yeah, it's a bit frustrating too.

File: plugins/@grouparoo/mixpanel/src/lib/test.ts (hunk 9 above)
Comment (pauloouriques) [reply plugins/@grouparoo/mixpanel/src/lib/test.ts:6]: Yes, it'll return [] (the current case, btw)

**Review by evantahler — APPROVED** [review]
Awesome!  
Looks like there are some merge conflicts now  - after that this is good to merge!

**Pull request merged**